src/scripts/main.js

```diff
@@ -1,3 +1,321 @@
+      ${message}
+    </p>
+  `;
+}
+
+function validate(key, value) {
+  if (!value) {
+    return false;
+  }
+
+  if (key === 'name' && value.length < 4) {
+    return false;
+  }
+
+  if (key === 'age' && ((value < 18 || value > 90) || !isFinite(value))) {
+    return false;
+  }
+
+  if (key === 'salary' && (value < 0 || !isFinite(value))) {
+    return false;
+  }
+
+  return true;
+}
+
+const table = document.querySelector('table');
+const tableHead = table.querySelector('thead');
+const tableBody = document.querySelector('tbody');
+
+initTableBody();
+initForm();
+
+const form = document.querySelector('form');
+const notification = document.querySelector('[data-qa="notification"]');
+
+let index;
+let count = 1;
+
+tableHead.addEventListener('click', (e) => {
+  const item = e.target.textContent.toLowerCase();
+
+  const dir = count % 2 === 0 && item === index
+    ? 'desc'
+    : 'asc';
+
+  usersData = usersData.sort((a, b) => {
+    let prev = a[item];
+    let curr = b[item];
+
+    if (isFinite(prev)) {
+      return dir === 'desc'
+        ? curr - prev
+        : prev - curr;
+    }
+
+    if (prev.includes('$')) {
+      prev = prev.replace(/[$,]/g, '');
+      curr = curr.replace(/[$,]/g, '');
+
+      return dir === 'desc'
+        ? curr - prev
+        : prev - curr;
+    }
+
+    return dir === 'desc'
+      ? curr.localeCompare(prev)
+      : prev.localeCompare(curr);
+  });
+
+  count++;
+  index = item;
+  initTableBody(usersData);
+});
+
+tableBody.addEventListener('click', (e) => {
+  for (const row of tableBody.children) {
+    row.classList.toggle('active', row === e.target.closest('tr'));
+  }
+});
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const userName = form.querySelector('[name="name"]');
+  const userAge = form.querySelector('[name="age"]');
+  const userOffice = form.querySelector('[name="office"]');
+  const userPosition = form.querySelector('[name="position"]');
+  const userSalary = form.querySelector('[name="salary"]');
+  const salary
+  = `$${userSalary.value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',')}`;
+  const formData = new FormData(form);
+
+  const removeNotification = () => {
+    notification.removeAttribute('class');
+    notification.innerHTML = '';
+  };
+
+  for (const [key, value] of formData) {
+    if (!validate(key, value)) {
+      initNotification('error', 'Please fill out all fields!');
+      setTimeout(removeNotification, 2000);
+
+      return false;
+    }
+
+    initNotification('success',
+      'Congratulations! Your data has been entered into the table!');
+  }
+
+  usersData.push({
+    name: userName.value,
+    position: userPosition.value,
+    office: userOffice.value,
+    age: userAge.value,
+    salary: salary,
+  });
+
+  initTableBody();
+  e.target.reset();
+  setTimeout(removeNotification, 2000);
+});
+
+tableBody.addEventListener('dblclick', (e) => {
+  const item = e.target.closest('td');
+  const i = item.cellIndex;
```

Comment:
```suggestion
  const columnIndex = item.cellIndex;
```
name i only works in for loops, otherwise it is almost impossible to understand what does it stand for. Try to press ctrl + F and try to find in which place your variable is used ;)